linkedList.js

```diff
@@ -3,6 +3,130 @@
  * @description  Main Module
  * @return {Object} an object exposing methods to be used to manipulate a linked list
  */
-function linkedListGenerator(){
+function linkedListGenerator() {
+  let head = null;
+  let tail = null;
 
-}
\ No newline at end of file
+  let getHead = function () {
+    return head;
+  };
+
+  let getTail = function () {
+    return tail;
+  }
+
+  let add = function (x) {
+    let newNode = {
+      value = x,
+      next = null
+    };
+
+    if (head === null) {
+      head = newNode;
+      tail = newNode;
+    } else {
+      tail.next = newNode;
+      tail = newNode;
+    }
+
+    return newNode;
+  }
+
+  let remove = function (x) {
+    if (x === 0) {
```

Comment:
This still does not handle the case if the linked list is empty and you try to remove something at index 0